sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -81,6 +89,141 @@ def __init__(
+            not name
+            or not namespace
+            or not storage_config
+            or not model_provider_parameters
+            or not dataset_provider_parameters
+            or not train_parameters
+            or not resources_per_worker
+        ):
+            raise ValueError("One of the required parameters is None")
+
+        try:
+            self.core_api.create_namespace(
+                body=utils.get_namespace_spec(namespace=namespace)
+            )
+        except Exception as e:
+            print(e)
+
+        PVC_NAME = "train-job-pvc"
+        self.core_api.create_namespaced_persistent_volume_claim(
+            namespace=namespace,
+            body=utils.get_pvc_spec(
+                pvc_name=PVC_NAME,
+                namespace=namespace,
+                storage_size=storage_config["size"],
+                storage_class=storage_config["storage_class"],
+            ),
+        )
+
+        if (
+            resources_per_worker["gpu"] is None and num_procs_per_worker != 0
+        ) or num_procs_per_worker > resources_per_worker["gpu"]:
+            raise ValueError("Insufficient gpu resources allocated to the container.")
+
+        if isinstance(model_provider_parameters, HuggingFaceModelParams):
+            mp = "hf"
+
+        if isinstance(dataset_provider_parameters, S3DatasetParams):
+            dp = "s3"
+
+        # create init container spec
+        init_container_spec = utils.get_container_spec(
+            name=constants.JOB_PARAMETERS[constants.PYTORCHJOB_KIND]["init_container"],
+            image=constants.JOB_PARAMETERS[constants.PYTORCHJOB_KIND][
+                "init_container_image"
+            ],
+            args=[
+                "--model_provider",
+                mp,
+                "--model_provider_parameters",
+                json.dumps(model_provider_parameters.__dict__),
+                "--dataset_provider",
+                dp,
+                "--dataset_provider_parameters",
+                json.dumps(dataset_provider_parameters.__dict__),
+            ],
+            volume_mounts=[
+                models.V1VolumeMount(name="train_job_pv", mount_path="/workspace")
+            ],
+        )
+
+        # create app container spec
+        container_spec = utils.get_container_spec(
+            name=constants.JOB_PARAMETERS[constants.PYTORCHJOB_KIND]["container"],
+            image=constants.JOB_PARAMETERS[constants.PYTORCHJOB_KIND][
+                "train_container_image"
+            ],
+            args=["--train_parameters", json.dumps(train_parameters.__dict__)],
+            volume_mounts=[
+                models.V1VolumeMount(name="train_job_pv", mount_path="/workspace")
+            ],
+            resources=models.V1ResourceRequirements(
+                limits={
+                    "nvidia.com/gpu": resources_per_worker["gpu"],
+                    "cpu": resources_per_worker["cpu"],
+                    "memory": resources_per_worker["memory"],
```

Comment:
Let's specify resources, similar to Katib SDK : https://github.com/kubeflow/katib/blob/master/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L321-L327.
E.g. maybe users want to run this API without GPUs.
Also, following this 🧵 : https://github.com/kubeflow/katib/pull/2192#discussion_r1284245434, we can set limits and requests to not confuse users.

cc @droctothorpe